kong/router/utils.lua

```diff
@@ -234,6 +234,133 @@ local function get_service_info(service)
          service_path
 end
 
+local phonehome_statistics
+do
+  local reports = require("kong.reports")
+  local nkeys = require("table.nkeys")
+  local worker_id = ngx.worker.id
+  local log = ngx.log
+  local ERR = ngx.ERR
+
+  local TILDE = byte("~")
+  local function is_regex_magic(path)
+    return byte(path) == TILDE
+  end
+
+  local empty_table = {}
+  -- reuse tables to avoid cost of creating tables and garbage collection
+  local protocols = {
+    http            = 0, -- { "http", "https" },
+    stream          = 0, -- { "tcp", "tls", "udp" },
+    tls_passthrough = 0, -- { "tls_passthrough" },
+    grpc            = 0, -- { "grpc", "grpcs" },
+  }
+  local path_handlings = {
+    v0 = 0,
+    v1 = 0,
+  }
+  local route_report = {
+    flavor = "unknown",
+    paths = 0,
+    headers = 0,
+    routes = 0,
+    regex_routes = 0,
+    protocols = protocols,
+    path_handlings = path_handlings,
+  }
+
+  local function traditional_statistics(routes)
+    local paths_count = 0
+    local headers_count = 0
+    local regex_paths_count = 0
+    local http = 0
+    local stream = 0
+    local tls_passthrough = 0
+    local grpc = 0
+    local v0 = 0
+    local v1 = 0
+
+    for _, r in ipairs(routes) do
+      r = r.route
+      local paths = r.paths or empty_table
+      local headers = r.headers or empty_table
+
+      paths_count = paths_count + #paths
+      headers_count = headers_count + nkeys(headers)
+      for _, path in ipairs(paths) do
+        if is_regex_magic(path) then
+          regex_paths_count = regex_paths_count + 1
+          break
+        end
+      end
+
+      for _, protocol in ipairs(r.protocols or empty_table) do -- luacheck: ignore 512
+        if protocol == "http" or protocol == "https" then
+          http = http + 1
+
+        elseif protocol == "tcp" or protocol == "tls" or protocol == "udp" then
+          stream = stream + 1
+
+        elseif protocol == "tls_passthrough" then
+          tls_passthrough = tls_passthrough + 1
+
+        elseif protocol == "grpc" or protocol == "grpcs" then
+          grpc = grpc + 1
+
+        else
+          log(ERR, "unknown protocol: " .. protocol)
```

Comment:
if this happens, user can do nothing, so let's not log error it. instead, consider sending an `unknown` in `protocols` table. we also have `ws` and `wss` in ee.